src/config/config.c

```diff
@@ -1341,22 +1341,29 @@ void readFTLconf(struct config *conf, const bool rewrite)
 	initConfig(conf);
 
 	// First try to read TOML config file
-	if(readFTLtoml(NULL, conf, NULL, rewrite, NULL))
+	// If we cannot parse /etc/pihole.toml (due to missing or invalid syntax),
+	// we try to read the rotated files in /etc/pihole/config_backup starting at
+	// the most recent one and going back in time until we find a valid config
+	for(unsigned int i = 0; i < PLAIN_ROTATIONS; i++)
 	{
-		// If successful, we write the config file back to disk
-		// to ensure that all options are present and comments
-		// about options deviating from the default are present
-		if(rewrite)
+		if(readFTLtoml(NULL, conf, NULL, rewrite, NULL, 0))
```

Comment:
```suggestion
		if(readFTLtoml(NULL, conf, NULL, rewrite, NULL, i))
```

You declared a counter `i` but it is not used anywhere. This seems a good place.